classes/Context.php

```diff
@@ -391,6 +392,15 @@ public function getTranslator($isInstaller = false)
         return $this->translator;
     }
 
+    /**
+     * @param string $locale
+     */
+    public function addLanguageToTranslator($locale)
```

Comment:
You're adding responsibilities to Context that aren't theirs.

I think that you should extract a class `TranslationLanguageLoader` that would basically take over the responsibility for loading a language into an already instantiated translator.

It would work like this:

```
$isAdminContext = defined('_PS_ADMIN_DIR_');
$translator = Context::getContext()->getTranslator();
(new TranslatorLanguageLoader($isAdminContext))->loadLanguage($translator, $localeCode); // this returns void
```

This way, I don't think that you'd need to change the `getTranslator()` signature anymore.